skyfield/almanac.py

```diff
@@ -135,6 +144,134 @@ def _find_maxima(start_time, end_time, f, epsilon=EPSILON, num=12):
+    Generally Earth.
+    topos: Topos - The passes will be calculated with respect to this
+    position.
+    satellite: EarthSatellite - Satellite to calculate passes of.
+    from_t and to_t: Time or datetime - Interval during which the passes will
+    be calculated. Default: from_t=now, to_t=24h from now
+    alt_deg_thresh: float - Degrees above the horizon. Will be calculated only
+    the passes that reach an altitude above this threshold
+
+    Returns a list of tuples each of which is of dimension 1x7 and contains,
+    respectively, the rising Time, the apex Time, the setting Time, the rising
+    azimut, the azimut at the apex, the setting azimut in degrees and the
+    altitude at the apex. The rising Time and the setting Time are based on the
+    set alt_deg_threshold.
+    """
+
+    # Get timestamps
+    if isinstance(from_t, Time):
+        from_ts = from_t.timestamp()
+    elif isinstance(from_t, datetime):
+        from_ts = datetime.timestamp(from_t)
+    else:
+        raise TypeError("Invalid type for from_t. Expected datetime or Time")
+
+    if isinstance(to_t, Time):
+        to_ts = to_t.timestamp()
+    elif isinstance(to_t, datetime):
+        to_ts = datetime.timestamp(to_t)
+    else:
+        raise TypeError("Invalid type for to_t. Expected datetime or Time")
+
+    # Find rise time and set time pairs
+    t, y = find_discrete(
+        ts.timestamp(from_ts),
+        ts.timestamp(to_ts),
+        satellite_visible(
+            ephemeris,
+            topos,
+            satellite,
+            alt_deg_thresh
+        ),
+        epsilon=1.1574074074074074e-05  # 1 second
+    )
+
+    rise_time = t[y]
+    set_time = t[~y]
+
+    azimuts_at_rise_time = [
+        satellite_azimut_at(
+            ephemeris,
+            topos,
+            satellite,
+            t
+        )
+        for t in rise_time
+    ]
+
+    azimuts_at_set_time = [
+        satellite_azimut_at(
+            ephemeris,
+            topos,
+            satellite,
+            t
+        )
+        for t in set_time
+    ]
+
+    rise_set_times = list(zip(rise_time, set_time))
+
+    # Find apices
+    apices_time = [
+        find_discrete(
+            _rise_time,
+            _set_time,
+            satellite_apices(
+                ephemeris,
+                topos,
+                satellite,
+                alt_deg_thresh
+            ),
+            epsilon=1.1574074074074074e-05  # 1 second
+        )[0][0]
+        for _rise_time, _set_time in rise_set_times
+    ]
+
+    azimuts_at_apex_time = [
+        satellite_azimut_at(
+            ephemeris,
+            topos,
+            satellite,
+            t
+        )
+        for t in apices_time
+    ]
+
+    altitudes = [
+        satellite_altitude_at(
+            ephemeris,
+            topos,
+            satellite,
+            t
+        )
+        for t in apices_time
+    ]
+
+    return list(
+        zip(
```

Comment:
Wow — does this work even if, say, the start time falls inside the time that the satellite is visible? I guess I should try it out with some sample values after the PR lands.

Reply:
I think it should

Reply:
It should be checked out anyway though. If the arrays don't match the extra elements get dropped and rise times get paired with previous rise_times

For example:
if rise_time = [rt2, rt3, ..., rtn]
and set_time = [st1, st2, ..., stn]

rt1 has passed so when zipping rt2 is paired with st1, rt3 with st2 and so on.

Perhaps the arrays should still be zipped in the end but the return could look like
```
return list(
    zip(
      rise_time,
      azimuts_at_rise_time
    )
  ),
  list(
    zip(
      apices_time,
      azimuts_at_apex_time,
      altitudes
    )
  ),
  list(
    zip(
      set_time,
      azimuts_at_set_time
    )
  )
```